src/submissions.ts

```diff
@@ -1,48 +1,44 @@
 import * as Db from './db';
 import {Platform, SourceCode} from './models';
 import {decodePlatformToken, PlatformTokenParameters} from './tokenization';
-import {getRandomId} from './util';
-
-export interface SubmissionParametersAnswer {
-    sourceCode?: string,
-    language?: string,
-}
-
-export interface SubmissionParametersUserTest {
-    name: string,
-    input: string,
-    output: string,
-}
-
-export interface SubmissionParametersTaskParameters {
-    returnUrl: string,
-}
-
-export interface SubmissionParameters {
-    token: string,
-    platform: string,
-    answer?: SubmissionParametersAnswer,
-    answerToken?: string,
-    taskId: string,
-    userTests?: SubmissionParametersUserTest[],
-    taskParams: SubmissionParametersTaskParameters,
-    sLocale: string,
-}
-
-export async function getPlatformTokenParams(token: string, platform: string, taskId: string): Promise<PlatformTokenParameters> {
+import {decode, getRandomId} from './util';
+import * as D from 'io-ts/Decoder';
+
+export const submissionDataDecoder = D.struct({
+  token: D.nullable(D.string),
+  platform: D.nullable(D.string),
+  answer: D.nullable(D.struct({
```

Comment:
Pay attention that optional (what you had before) is not the same as nullable. So you changed your format in practice.

Reply:
Thanks for the info, I hadn't realized!